File: yt/yt/flow/examples/java/wait_click_join/wait_click_join/src/main/java/tech/ytsaurus/flow/examples/waitclickjoin/PipelineConfiguration.java
package tech.ytsaurus.flow.examples.waitclickjoin;

import java.util.List;

import org.springframework.context.annotation.Configuration;
import tech.ytsaurus.flow.examples.waitclickjoin.model.Action;
import tech.ytsaurus.flow.examples.waitclickjoin.model.Hit;
import tech.ytsaurus.flow.examples.waitclickjoin.model.JoinedAction;
import tech.ytsaurus.flow.spring.ComputationProvider;
import tech.ytsaurus.flow.stream.FlowStream;
import tech.ytsaurus.flow.stream.FlowStreams;

// [BEGIN pipeline_configuration]
@Configuration
public class PipelineConfiguration implements ComputationProvider {

    @Override
    public List<FlowStream<?>> getStreams() {
        return List.of(
                FlowStreams.typed("hit", Hit.class),
                FlowStreams.typed("action", Action.class),
                FlowStreams.typed("joined_action", JoinedAction.class)
        );
    }
}
// [END pipeline_configuration]

File: yt/yt/flow/examples/kotlin/wait_click_join/wait_click_join/src/main/kotlin/tech/ytsaurus/flow/examples/waitclickjoin/PipelineConfiguration.kt
package tech.ytsaurus.flow.examples.waitclickjoin

import org.springframework.context.annotation.Configuration
import tech.ytsaurus.flow.computation.Computation
import tech.ytsaurus.flow.examples.waitclickjoin.model.Action
import tech.ytsaurus.flow.examples.waitclickjoin.model.Hit
import tech.ytsaurus.flow.examples.waitclickjoin.model.JoinedAction
import tech.ytsaurus.flow.spring.ComputationProvider
import tech.ytsaurus.flow.stream.FlowStream
import tech.ytsaurus.flow.stream.FlowStreams

// [BEGIN pipeline_configuration]
@Configuration
open class PipelineConfiguration : ComputationProvider {
    override fun getComputations(): List<Computation> {
        val join = Computation.builder()
            .setComputationId("join")
            .setProcessFunction(JoinProcessFunction())
            .build()
        return listOf(join)
    }

    override fun getStreams(): List<FlowStream<*>> {
        return listOf(
            FlowStreams.typed("hit", Hit::class.java),
            FlowStreams.typed("action", Action::class.java),
            FlowStreams.typed("joined_action", JoinedAction::class.java)
        )
    }
}
// [END pipeline_configuration]
